### PR TITLE
Initial Dashboard and finalizing login page

### LINKED ISSUE #32: UI - Dashboard (Campaign selector)

### PR DESCRIPTION
**Pull request by KarmaKamikaze**
**_This is a UI enhancement PR and thus should also be reviewed on the webpage in a running state in addition to the source code._**

**Summary of added content:**
- New unauthorized index page with typing logo and fading slogan.
  - Cool and immersive background image on index page.
  - Login and register buttons moved to a central location.
  - Smooth animated buttons that gives off that 'boomer tryin'a be zoomer' kinda vibes.
  - Removed navbar in top and side of screen since login functionality has moved to a central position.
- New Dashboard page for authorzed users.
  - Tweaked navbar for our ChatRPG branding, allowing for logging out as well (I know, amazing stuff).
  - List box with previously played campaigns which players can choose to continue.
  - Area for creating a brand new campaign
    - Lets players choose a Campaign title, a Character name, and a custom starting scenario.
    - Prewritten starting scenarios can also chosen and adapted by the user.
- Adds the ChatRPGTests project for unit/E2E tests

**Relevant files for review:**
- `ChatRPG/Pages/Index.razor`
- `ChatRPG/Pages/Index.razor.cs`
- `ChatRPG/Shared/LoginDisplay.razor`
- `ChatRPG/Shared/MainLayout.razor`
- `ChatRPG/Pages/UserCampaignOverview.razor`
- `ChatRPG/Pages/UserCampaignOverview.razor.cs`
- `ChatRPG/Pages/Shared/Navbar.razor`

This pull request closes #32

### CODE REVIEW

File: ChatRPGTests/ChatRPGFixture.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            }
+            catch (HttpRequestException ex)
+            {
+                Console.WriteLine($"Request failed: {ex.Message}");
+            }
+
+            Task.Delay(delayBetweenAttempts).Wait();
+        }
+
+        throw new InvalidOperationException("Failed to determine if the app is ready.");
```
Comment (damniko) [inline ChatRPGTests/ChatRPGFixture.cs:62]: Er det her den smarteste måde at teste ASP.NET apps på?
Har ikke erfaring med det, så det er et reelt spørgsmål. Jeg forestiller mig bare at der må være nogle bedre værktøjer til det.

**Review by damniko — APPROVED** [review]

File: ChatRPGTests/ChatRPGFixture.cs (hunk 1 above)
Comment (KarmaKamikaze) [reply ChatRPGTests/ChatRPGFixture.cs:62]: Det er ikke en test, det er bare en måde at sikre os at vores server application kører på inden vores tests kører. Det er den smarteste måde at gøre det på, medmindre du vil gøre som alle de andre plebs og køre vores application manuelt hver gang der køres E2E test cases.

File: ChatRPG/Pages/Index.razor [hunk 2]
```diff
@@ -1,9 +1,33 @@
...
+            </h1>
+            <p class="slogan-front text-readable">Immerse yourself in the ultimate AI-powered adventure!</p>
+            <LoginDisplay/>
+        </div>
+    </NotAuthorized>
+    <Authorized>
+        <PageTitle>Dashboard - ChatRPG</PageTitle>
 
-Welcome to your new app.
+        <div class="dashboard-wrapper">
```
Comment (selectionator) [inline ChatRPG/Pages/Index.razor:18]: "Er den centered?" - Sarmi

**Review by selectionator — APPROVED** [review]
Ser sgu nice ud! LGTM :+1:

File: ChatRPG/Pages/Index.razor (hunk 2 above)
Comment (KarmaKamikaze) [reply ChatRPG/Pages/Index.razor:18]: image

**Review by Sarmisuper — APPROVED** [review]

**Pull request merged**